app/services/CompileService.scala

```diff
@@ -36,38 +35,40 @@ class CompileService @Inject()(
                                 actorSystem: ActorSystem
                               ) (implicit ec: ExecutionContext) {
 
-
   private def doCompile(currentVersionInfo: VersionInfo,
-                newVersionInfo: VersionInfo,
-                pstr: String,
-                userAnswers: UserAnswers,
-                delete :Boolean,
-                compileResponse: Future[Unit])(implicit headerCarrier: HeaderCarrier): Future[Unit] = {
+                        newVersionInfo: VersionInfo,
+                        pstr: String,
+                        userAnswers: UserAnswers,
+                        delete: Boolean,
+                        eventType: Option[EventType],
+                        deleteAttr: Option[(String, EventDataIdentifier, Int, String)])(implicit headerCarrier: HeaderCarrier): Future[Unit] = {
 
     val futureOptChangedOverviewSeq = if (newVersionInfo.version > currentVersionInfo.version) {
       updateUAVersionAndOverview(pstr, userAnswers, currentVersionInfo.version, newVersionInfo.version)
     } else {
       Future.successful(None)
     }
+
     futureOptChangedOverviewSeq.flatMap { optChangedOverviewSeq =>
       val uaNonEventTypeVersionUpdated = userAnswers.setOrException(VersionInfoPage, newVersionInfo, nonEventTypeData = true)
       val updatedUA = optChangedOverviewSeq match {
         case Some(seqErOverview) => uaNonEventTypeVersionUpdated
           .setOrException(EventReportingOverviewPage, seqErOverview, nonEventTypeData = true)
         case _ => uaNonEventTypeVersionUpdated
       }
-      userAnswersCacheConnector.save(pstr, updatedUA).flatMap { _ =>
-        val response = compileResponse
-        response.map { _ =>
-          appConfig.compileDelayInSeconds match {
```

Comment:
I've noticed that appConfig.compileDelayInSeconds was removed. Should we add this back in?

We have previously noticed while testing on the QA environment. After submitting a compile, even tho we got the response from IF confirming that the request have been processed, there was a delay before the data on the API was updated.

What happened is that, the next page right after compile would be served without the last update, so on list of Events page the updated event would be missing. The IF endpoint is essentially misconfigured and that's our way of dealing with it. It supposed to be synchronous API endpoint, but it's not, and there is no callback function provided by IF.

Also, let me know if this is no longer relevant and maybe we've misidentified the problem.

Reply:
Pavel, i guess       userAnswersCacheConnector.save(pstr, updatedUA).flatMap(_ => futureAction)
 this line will make sure the second future start executing only after first future returned succesfully. so i guess no need to wait.

Reply:
Can we test this in QA @vikramsvr should know how to test.